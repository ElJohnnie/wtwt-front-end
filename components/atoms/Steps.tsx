import styles from 'styles\Steps.modules.css'

interface StepsProps{
    number: number;
    color: string;
}

export default function Steps(props: StepsProps) {
    return <div>

    </div>
}