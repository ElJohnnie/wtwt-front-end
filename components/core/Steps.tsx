import React from 'react';

export interface StepsProps {
  number: string;
  hasRange: boolean;
  isEmpty?: boolean;
  asComplete?: boolean;
  isInactive: boolean;
}

export default function Steps(props: StepsProps) {
  const { number, hasRange, asComplete, isEmpty, isInactive } = props;
  return (
    <div className="flex flex-row justify-center items-center">
      <div
        className={`size-10 md:size-16 xl:size-18 rounded-full  
          inline-flex items-center justify-center  
          text-white text-[24px] font-bold ${
            isInactive ? 'bg-pink-400' : 'bg-pink-600'
          }`}
      >
        {number}
      </div>
      {hasRange && (
        <div className="w-8 lg:w-24 md:w-16 bg-gray-200 rounded-full invisible md:visible mx-0 h-2.5 md:mx-4 dark:bg-gray-700">
          <div
            className="bg-pink-600 h-2.5 rounded-full"
            style={
              !isEmpty && !isInactive
                ? { width: asComplete ? '100%' : '50%' }
                : { width: '0%' }
            }
          ></div>
        </div>
      )}
    </div>
  );
}
