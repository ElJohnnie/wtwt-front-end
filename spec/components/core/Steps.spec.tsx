import React from 'react';
import { render } from '@testing-library/react';
import Steps from '../../../components/core/Steps';

describe('Steps', () => {
  it('quero renderizar o component de steps inicialmente', () => {
    const { container } = render(
      <Steps
        number={'1'}
        hasRange={false}
        asComplete={false}
        isEmpty={false}
        isInactive={false}
      />,
    );
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
  it('quero renderizar o component de steps inativo', () => {
    const { container } = render(
      <Steps
        number={'1'}
        hasRange={false}
        asComplete={false}
        isEmpty={false}
        isInactive={true}
      />,
    );
    expect(container).toBeTruthy();
  });
  it('quero renderizar o component de steps com o range completo', () => {
    const { container } = render(
      <Steps
        number={'1'}
        hasRange={true}
        asComplete={true}
        isEmpty={false}
        isInactive={false}
      />,
    );
    expect(container).toBeTruthy();
  });
  it('quero renderizar o component de steps com o range incompleto', () => {
    const { container } = render(
      <Steps
        number={'1'}
        hasRange={true}
        asComplete={false}
        isEmpty={false}
        isInactive={false}
      />,
    );
    expect(container).toBeTruthy();
  });
  it('quero renderizar o component de steps com o range vazio', () => {
    const { container } = render(
      <Steps
        number={'1'}
        hasRange={true}
        asComplete={false}
        isEmpty={true}
        isInactive={false}
      />,
    );
    expect(container).toBeTruthy();
  });
});
