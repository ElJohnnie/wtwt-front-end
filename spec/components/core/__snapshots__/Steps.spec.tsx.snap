// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps quero renderizar o component de steps inicialmente 1`] = `
<div>
  <div
    class="flex flex-row justify-center items-center"
  >
    <div
      class="size-10 md:size-16 xl:size-18 rounded-full  
          inline-flex items-center justify-center  
          text-white text-[24px] font-bold bg-pink-600"
    >
      1
    </div>
  </div>
</div>
`;
